listenbrainz/webserver/views/profile.py

```diff
@@ -242,3 +244,48 @@ def request_stats():
         _redis.redis.set(construct_stats_queue_key(current_user.musicbrainz_id), 'queued')
         flash.info('You have been added to the stats calculation queue! Please check back later.')
     return redirect(url_for('profile.info'))
+
+
+@profile_bp.route('/delete', methods=['GET', 'POST'])
+@login_required
+def delete():
+    """ Delete currently logged-in user from ListenBrainz.
+
+    If POST request, this view checks for the correct authorization token and
+    deletes the user. If deletion is successful, redirects to home page, else
+    flashes an error and redirects to user's info page.
+
+    If GET request, this view renders a page asking the user to confirm
+    that they wish to delete their ListenBrainz account.
+    """
+    if request.method == 'POST':
+        if request.form.get('token') == current_user.auth_token:
+            try:
+                delete_user(current_user.musicbrainz_id)
+            except Exception as e:
+                flash.error('Error while deleting user %s, please try again later.' % current_user.musicbrainz_id)
+                return redirect(url_for('profile.info'))
+            return redirect(url_for('index.index'))
+        else:
+            flash.error('Cannot delete user due to error during authentication, please try again later.')
+            return redirect('profile.info')
+    else:
+        return render_template('profile/delete.html', token=current_user.auth_token)
+
+
+def delete_user(musicbrainz_id):
+    """ Delete a user from ListenBrainz completely.
+    First, drops the user's influx measurement and then deletes her from the
+    database.
+
+    Args:
+        musicbrainz_id (str): the MusicBrainz ID of the user
+
+    Raises:
+        NotFound if user isn't present in the database
+    """
+
+    #TODO(param): delete user's listens from Google BigQuery
```

Comment:
I think this is a job we should push into stats_calculator (might have to rename that to something that indicates generic bigquery operation executor)?

Reply:
[LB-355](https://tickets.metabrainz.org/browse/LB-355)